Code/Uplynk/Source/Network/UplynkApi.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  UplynkAPI.swift
+//
+//  Created by Raffi on 09/09/2024.
+//  Copyright © 2024 THEOplayer. All rights reserved.
+//
+
+import Foundation
+
+
+protocol UplynkAPIProtocol {
+    static func requestLive(preplaySrcURL: String) async -> PrePlayLiveResponse?
+    static func requestVOD(preplaySrcURL: String) async -> PrePlayVODResponse?
+}
+
+class UplynkAPI: UplynkAPIProtocol {
+    
+    static func requestLive(preplaySrcURL: String) async -> PrePlayLiveResponse? {
+        do {
+            let data = try await HTTPSConnection.request(type: .get, urlString: preplaySrcURL)
+            let decoder: JSONDecoder = .init()
+            guard let preplayResponse: PrePlayLiveResponse = try? decoder.decode(PrePlayLiveResponse.self, from: data) else {
```

Comment:
Testing a live response using the `live` asset id's in the sample app, it fails decode properly?

```
keyNotFound(CodingKeys(stringValue: "ads", intValue: nil), Swift.DecodingError.Context(codingPath: [], debugDescription: "No value associated with key CodingKeys(stringValue: \"ads\", intValue: nil) (\"ads\").", underlyingError: nil))
```

Reply:
ah right. I'll look into it. I haven't tested live assets yet because they were not available without DRM